packages/flutter_tools/lib/runner.dart

```diff
@@ -128,18 +132,52 @@ Future<int> _handleToolError(
       command: args.join(' '),
     );
 
-    if (error is String) {
-      stderr.writeln('Oops; flutter has exited unexpectedly: "$error".');
-    } else {
-      stderr.writeln('Oops; flutter has exited unexpectedly.');
-    }
+    final String command = _crashCommand(args);
+
+    final String errorString = error is String ? error : error.toString();
+    printError(errorString);
 
     try {
-      final File file = await _createLocalCrashReport(args, error, stackTrace);
-      stderr.writeln(
-        'Crash report written to ${file.path};\n'
-            'please let us know at https://github.com/flutter/flutter/issues.',
-      );
+      final String doctorText = await _doctorText();
+      final File file = await _createLocalCrashReport(args, error, stackTrace, doctorText);
+
+      printError('A crash report has been written to ${file.path}.');
+      printStatus('This crash may already be reported. Check GitHub for similar crashes.', emphasis: true);
+
+      printStatus('https://github.com/flutter/flutter/issues?q=is%3Aissue+${Uri.encodeQueryComponent('$errorString')}\n', wrap: false);
+      printStatus('To report your crash to the Flutter team, first read the guide to filing a bug.', emphasis: true);
+      printStatus('https://flutter.dev/docs/resources/bug-reports\n', wrap: false);
+      printStatus('Create a new GitHub issue. Paste this link into your browser and complete the template. Thank you!', emphasis: true);
+
+      final String title = '[tool_crash] $errorString';
+      final String body = '''## Command
+```
+$command
+```
+
+## Steps to Reproduce
+1. ...
+2. ...
+3. ...
+
+## Logs
+${_crashException(error)}
+```
+${LineSplitter.split(stackTrace.toString()).take(20).join('\n')}
+```
+```
+$doctorText
+```
+
+## Flutter Application Metadata
+${_projectMetadataInformation()}
+''';
+      printStatus('https://github.com/flutter/flutter/issues/new?title=${Uri.encodeQueryComponent(title)}'
+        '&body=${Uri.encodeQueryComponent(body)}'
+        '&labels=${Uri.encodeQueryComponent('tool,severe: crash')}\n', wrap: false);
+
+      printStatus('Oops; flutter has exited unexpectedly.', emphasis: true, color: TerminalColor.red);
```

Comment:
I wanted this to be `printError` but when I changed it to that it always printed before some of the above status lines.  I'm guessing stderr has more aggressive flush behavior, and none of the stdout flushes I tried seemed to help.  Open to suggestions.

Reply:
Unfortunately it looks like `dart:io` `stdout.flush()` only flushes any buffering in Dart's memory. It doesn't actually result in a `flush()` call on the underlying fd to the OS.

Reply:
I'm okay with an angry-looking red status.

Reply:
I switched this back to an error at the top since now the output is much smaller and I'm less worried about it getting scrolled out of view.